ast/compile.go

```diff
@@ -883,6 +883,47 @@ func compileSchema(goSchema interface{}) (*gojsonschema.Schema, error) {
 	return schemasCompiled, nil
 }
 
+func mergeSchemas(schemas []*gojsonschema.SubSchema) (*gojsonschema.SubSchema, error) {
+	if len(schemas) == 0 {
+		return nil, nil
+	}
+	var result = schemas[0]
+
+	for i := 0; i < len(schemas); i++ {
+		if len(schemas[i].PropertiesChildren) > 0 {
+			if !schemas[i].Types.Contains("object") {
+				if err := schemas[i].Types.Add("object"); err == nil {
+					schemas[i].Types.Add("object")
+				}
```

Comment:
Actually, the check needs to for `err != nil` <-- thats the concerning case.
The first Add (inside the if) already adds the the object type so we dont need to do it twice.

So like this:
```
if err := schemas[i].Types.Add("array"); err != nil {
	return nil, fmt.Errorf("unable to set the type in schemas")
}
```